doc/architecture/software-update.rst

```diff
@@ -17,15 +17,46 @@ Ostro OS based software can be deployed to a target device in two ways:
 - Software Update
 
   This is a component that Ostro OS borrows from Clear Linux\* OS
-  for Intel |reg| Architecture and consists of both a server and client
-  component.
+  and consists of both a server and client component.
+  While Clear Linux is focused on Intel |reg| Architecture, the
+  Software Update component is not tied to any specific platform
+  nor architecture, as long as the concept of rootfs/filesystem is
+  supported.
+
+
+Software Update
+===============
+The main concept behind the `Software Update tool`_ is that it doesn't
+rely on the target device to perform the installation of packages,
+handle file conflicts, dependency and versioning check.
+Instead, all of this happens a priori, on a build server.
+And it happens only once, instead of trying to repeat the same
+operations on each device.
+All the device needs to do is to lookup a reference file (manifest)
+for the list of files it must acquire, should they not be present locally.
+The tool focuses on high level functionality (bundles), instead of
+individual packages.
+In detail, the Software Update functionality is comprised of a pair of
+tools:
+
+ - Software Update Server
```

Comment:
looks a bit better if you don't indent these bullet items (bring these two bullet lines back to to the left margin)